src/Tokenizer/Transformer/BraceTransformer.php

```diff
@@ -145,8 +157,40 @@ private function transformIntoDynamicVarBraces(Tokens $tokens, Token $token, int
         $tokens[$closeIndex] = new Token([CT::T_DYNAMIC_VAR_BRACE_CLOSE, '}']);
     }
 
-    private function transformIntoCurlyIndexBraces(Tokens $tokens, Token $token, int $index): void
+    private function transformIntoPropertyHookBraces(Tokens $tokens, int $index): void
     {
+        if (\PHP_VERSION_ID < 8_04_00) {
```

Comment:
>  the fixes are given the the enabled rules.

hey, would you mind to rephrase?

Reply:
typo, sorry, ...given by...

Reply:
transformer is executed regardless of enabled fixers, if you refer to that

Reply:
I refer to... The attribute tokens should be transformed no matter the PHP runtime. In ideal case, the PHP CS Fixer should be possible to run on PHP 7.4 to fix files even with PHP 8.4 syntax.

Reply:
Then i find this comment totally out of scope of this PR.

If you want PHP CS Fixer on 7.4 to understand and process files written in 8.4 syntax, you need to change the tokenizer from one built-in into PHP itself to sth custom/forked etc. Otherwise, we are binding runtime PHP with version of PHP target file was written for.

This would be a massive change with huge impact and I'm not gonna go this path.
As it was discussed few times in the repo already, I will not engage deeper into this discussion. If you want to convince me to change my decision, please send a PR where you remove all `* @requires PHP x.y` from testbase and yet have them properly passing.

Reply:
I know this limitations. The question is can this transformer work since PHP 8.0 where attributes were introduced?

Reply:
this PR touches property hooks (8.4) and not attributes

if you would want to apply it for 8.0 regardless of that - the code won't be executed on 8.0, because 8.4-syntax will be consider invalid on low-level, built-in tokenizer and no transformer will even have chance to execute